### PR TITLE
MenuGenerator raises a right error when a menu item is not found #134

### PR DESCRIPTION
**Pull request by LukasRychtecky**
MenuGenerator checks if Site contains an iscore instance
before getting it.

### CODE REVIEW

File: is_core/menu.py [hunk 1]
```diff
@@ -43,12 +43,18 @@ def get_menu_items(self, items):
         for item in items:
             if isinstance(item, MenuItem):
                 menu_items.append(item)
-            else:
-                item = self.site._registry[item]
+            elif self.site.has_iscore(item):
```
Comment (LukasRychtecky) [inline is_core/menu.py]: it should check if site has an iscore

File: is_core/menu.py [hunk 2]
```diff
@@ -43,12 +43,18 @@ def get_menu_items(self, items):
...
+                item = self.site.get_iscore(item)
                 menu_item = item.get_menu_item(self.request, group)
                 if menu_item:
                     menu_items.append(menu_item)
+            else:
+                raise MenuError('IS Core or Menu item not found for name "%s"' % item)
```
Comment (LukasRychtecky) [inline is_core/menu.py]: Raises more friendly error than KeyError

File: is_core/menu.py [hunk 3]
```diff
@@ -43,12 +43,18 @@ def get_menu_items(self, items):
...
         return menu_items
 
     def get_menu_structure(self):
-        return self.site._registry.keys()
+        return self.site.get_iscore_names()
```
Comment (LukasRychtecky) [inline is_core/menu.py]: an internal field should not be used from outside

File: is_core/site.py [hunk 4]
```diff
@@ -42,6 +42,17 @@ def __init__(self, name='IS'):
         sites[name] = self
         self._registry = self._init_items()
 
+    def has_iscore(self, name):
+        return name in self._registry
+
+    def get_iscore(self, name):
+        if not self.has_iscore(name):
+                raise ImproperlyConfigured('IS Core "%s" not found' % name)
```
Comment (matllubos) [inline is_core/site.py]: bad indent

File: is_core/site.py [hunk 5]
```diff
@@ -42,6 +42,17 @@ def __init__(self, name='IS'):
         sites[name] = self
         self._registry = self._init_items()
 
+    def has_iscore(self, name):
```
Comment (matllubos) [inline is_core/site.py]: maybe has_registered_is_core ? or is_registered_is_core

File: is_core/site.py [hunk 6]
```diff
@@ -42,6 +42,17 @@ def __init__(self, name='IS'):
...
+        return name in self._registry
+
+    def get_iscore(self, name):
```
Comment (matllubos) [inline is_core/site.py]: please split is ando core, so is_core

File: is_core/site.py [hunk 7]
```diff
@@ -42,6 +42,17 @@ def __init__(self, name='IS'):
...
+        if not self.has_iscore(name):
+                raise ImproperlyConfigured('IS Core "%s" not found' % name)
+        return self._registry[name]
+
+    def get_iscore_names(self):
```
Comment (matllubos) [inline is_core/site.py]: get_is_core_names

File: is_core/menu.py (hunk 3 above)
Comment (matllubos) [inline is_core/menu.py]: good point

File: is_core/menu.py (hunk 2 above)
Comment (matllubos) [inline is_core/menu.py]: good point

File: is_core/menu.py [hunk 8]
```diff
@@ -43,12 +43,18 @@ def get_menu_items(self, items):
...
+
+
+class MenuError(Exception):
```
Comment (matllubos) [inline is_core/menu.py]: we have exceptions package for errors in django-is-core, but maybe ImproperlyConfigured error will be enought. There is no reason to create custom exception

**Comment by LukasRychtecky** [comment]
Fixed

**Pull request closed without merging**